internal/storage/serde.go

```diff
@@ -103,6 +103,8 @@ type serdeEntry struct {
 	serialize func(array.Builder, any) bool
 	// sizeof returns the size in bytes of the value
 	sizeof func(any) uint64
+	// encoding returns the encode method if parquet dictionary encoding is disabled
+	encoding func() parquet.Encoding
```

Comment:
IMO the best encoding is determined by the data distribution, rather than data type.

According to below code, it looks more like a `primaryKeyEncoding`?